src/index.test.ts

```diff
@@ -120,6 +120,24 @@ describe('jsonFetch', function () {
       expect(response.statusText).to.equal('Created');
       expect(response.headers).to.be.ok();
     });
+
+    it('calls onRequestStart when is passed', async function () {
+      const onRequestStart = sandbox.spy(() => {});
```

Comment:
nitpick: no reason to make this a spy, I think?
```suggestion
      const onRequestStart = sandbox.stub();
```